sympy/core/numbers.py

```diff
@@ -438,6 +439,142 @@ def ilcm(*args):
     return a
 
 
+def igcdLLL(s, m1=2, n1=3, optimize=False):
+    """return g, B where g is the gcd of the integers in s and the
+    LLL-reduced basis and small multipliers such that those
+    multiples of the corresponding values of s is g.
+
+    If there are three values in s then the smallest multipliers
+    will be returned unless optimize is not True.
+
+    This is an iterative procedure whose quality in returning
+    small multipliers is related to alpha = m1/n1 which must
+    be in the range [3/8, 1]. When 1, the algorithm will iterate
+    longer but is not guaranteed to return the smallest multipliers
+    except for the case of 3 values in s.
+
+    Examples
+    ========
+
+    >>> from sympy.core.numbers import igcdLLL
+    >>> from sympy import Add, Matrix, symbols
+
+    >>> s = 2, 4, 6
+    >>> x = symbols('x:%s' % len(s))
+    >>> eq = Add(*[si*xi for si, xi in zip(s, x)]); eq
+    2*x0 + 4*x1 + 6*x2
+
+    >>> g, B = igcdLLL(s)
+    >>> assert Add(*[si*mi for si, mi in zip(s, B[-1])]) == g
+
+    >>> p = symbols('p:%s' % (len(B) - 1))
+    >>> psol = (Matrix(B[:-1]).T*Matrix(p)).T.tolist()[0]
+    >>> dict(zip(x, psol))
+    {x0: p0 + 2*p1, x1: p0 - p1, x2: -p0}
+    >>> eq.subs(_)
+    0
+    """
+    if not iterable(s):
+        raise ValueError('s should be iterable')
+    if len(s) < 2:
+        raise ValueError('expecting 2 or more values')
+    if not all(type(i) is int for i in s):
+        raise ValueError('expecting literal ints for s')
+    ok = (3*n1 <= 8*m1) and (n1 > 0) and (m1 > 0) and (m1 <= n1)
+    if not ok:
+        raise ValueError('require 3/8 <= alpha <= 1 but alpha = %s/%s' % (m1, n1))
+    if optimize and len(s) > 3:
+        raise ValueError('optimization for more than 3 values')
+
+    def reduce1(k, i):
+        if a[i] != 0:
+            q = int(round(a[k] / a[i]))
+        else:
+            if 2 * abs(la[k][i]) > D[i]:
+                q = int(round(la[k][i] / D[i]))
+            else:
+                q = 0
+        if q != 0:
+            a[k] -= q * a[i]
+            for c in range(m):
+                B[k][c] -= q * B[i][c]
+            la[k][i] -= q * D[i]
+            for j in range(i):
+                la[k][j] -= q * la[i][j]
+
+    def swap(k):
+
+        # swap rows k and k - 1 and calculate new la and D
+
+        j = k - 1
+        (a[k], a[j]) = (a[j], a[k])
+        (B[k], B[j]) = (B[j], B[k])
+        for i in range(j):
+            (la[k][i], la[j][i]) = (la[j][i], la[k][i])
+        for i in range(k + 1, m):
+            t = la[i][j] * D[k] - la[i][k] * la[k][j]
+            la[i][j] = (la[i][j] * la[k][j] + la[i][k] * D[j - 1]) \
+                / D[j]
```

Comment:
It's better to have a line that is slightly longer than to wrap like this.

Reply:
`black` agrees